### PR TITLE
Improve ping API a bit by returning failure in case of only failure

### PR DESCRIPTION
**Pull request by Kubuxu**
Part of https://github.com/ipfs/go-ipfs/issues/2803
License: MIT
Signed-off-by: Jakub Sztandera <kubuxu@protonmail.ch>

### CODE REVIEW

File: core/commands/ping.go [hunk 1]
```diff
@@ -131,14 +132,18 @@ func pingPeer(ctx context.Context, n *core.IpfsNode, pid peer.ID, numPings int)
...
+		}
 
 		ctx, cancel := context.WithTimeout(ctx, kPingTimeout*time.Duration(numPings))
 		defer cancel()
 		pings, err := n.Ping.Ping(ctx, pid)
 		if err != nil {
 			log.Debugf("Ping error: %s", err)
-			outChan <- &PingResult{Text: fmt.Sprintf("Ping error: %s", err)}
+			outChan <- &PingResult{
+				Text: fmt.Sprintf("Ping error: %s", err)}
```
Comment (whyrusleeping) [inline core/commands/ping.go]: lets explicitly set Success false here (for clarity to the reader), and then put the closing bracket on its own line.

**Comment by Kubuxu** [comment]
Updated.

**Comment by jbenet** [comment]
Not quite sure what this does-- the PR title doesn't tell me, and the code is not clear why-- maybe add some comments why some return success and one fail?

**Comment by Kubuxu** [comment]
We have boolean filed `Success` which as boolean fields default to false, was implying failure in NDJSON messages that didn't fail. Now, we explicitly set success to true on all of them and to false if there is failure.

**Pull request merged**